docs/next/link: Mention middleware + dynamic routes changes (#42607)
vercel/next.js: docs/api-reference/next/link.md

  Disables scrolling to the top
</Link>
```

## With Next.js 13 Middleware

> It's common to use [Middleware](/docs/advanced-features/middleware) for authentication or other purposes that involve redirecting the user to a different page. In versions prior to Next.js 13, the redirect is handled on the client instead of the server. So, in order for the `<Link />` component to properly prefetch links with redirects via Middleware, you need to tell Next.js both the URL to display and the URL to prefetch.

Reviewer: ```suggestion
It's common to use [Middleware](/docs/advanced-features/middleware) for authentication or other purposes that involve rewriting the user to a different page. In order for the `<Link />` component to properly prefetch links with rewrites via Middleware, you need to tell Next.js both the URL to display and the URL to prefetch. This is required to avoid un-necessary fetches to middleware to know the correct route to prefetch. 
```

For example, if you have want to serve a `/dashboard` route that has authenticated and visitor views, you may add something similar to the following in your Middleware to redirect the user to the correct page:

> ```js:middleware.js

Reviewer: ```suggestion
```js
// middleware.js
```

export function middleware(req) {
  const nextUrl = req.nextUrl;
  if (nextUrl.pathname === '/dashboard') {
    if (req.cookies.authToken) {
      return NextResponse.rewrite('/auth/dashboard');
    } else {
      return NextResponse.rewrite('/public/dashboard');
    }
  }
}
```

In this case, you would want to use the following code in your `<Link />` component (inside `pages/`):

> ```js:pages/index.js

Reviewer: ```suggestion
```js
// pages/index.js
```
